src/components/signature/signature.test.tsx

```diff
@@ -0,0 +1,58 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+import Signature from './signature';
+
+const testData = [
+  {
+    author: 'John Doe',
+    date: '2020-07-01',
+  },
+  {
+    author: 'John Doe',
+    date: new Date().toString(),
+  },
+  {
+    author: 'John Doe',
+    date: '2020-08-01',
+  },
+];
+
+describe(Signature.name, () => {
+  it('should render', () => {
+    const signature = render(
+      <Signature author={testData[0].author} date={testData[0].date} />,
+    );
+    expect(signature).toBeTruthy();
+  });
+
+  it('should contain author', () => {
+    const { getByText } = render(
+      <Signature author={testData[0].author} date={testData[0].date} />,
+    );
+    expect(getByText('von John Doe')).toBeTruthy();
+  });
+
+  describe('days label', () => {
+    it('should render label "Tag" if date today', () => {
+      const { findByText } = render(
+        <Signature author={testData[1].author} date={testData[1].date} />,
+      );
+      expect(findByText('Tag')).toBeTruthy();
+    });
+
+    it('should render label "Tagen" if date earlier than today', () => {
+      const { findByText } = render(
+        <Signature author={testData[0].author} date={testData[0].date} />,
+      );
+      expect(findByText('Tagen')).toBeTruthy();
+    });
+
+    it('should render correct number of days', () => {
+      const { getByText } = render(
+        <Signature author={testData[2].author} date={testData[2].date} />,
+      );
+      const date = getByText('vor 5 Tagen');
+      expect(date).toBeTruthy();
```

Comment:
This test will only work today ;-) 

You can use the calculation method you already use in the component to get the number of days, and check if the component returns the same amount.

Reply:
oh I missed that 😬 Actually I would mock the Date.now to a specific date. Date.now = jest.fn(() => new Date(Date.UTC(2000, 0, 1)).valueOf());

Reply:
Can also be done like this, yep. Aditionally, I would put it into `beforeEach` and unmock it in `afterEach` to prevent nasty side effects.
https://github.com/facebook/jest/issues/2234#issuecomment-324868057